controller/src/main/java/io/pravega/controller/server/ControllerServiceStarter.java

```diff
@@ -286,14 +305,23 @@ protected void startUp() {
             // Wait for controller event processors to start.
             if (serviceConfig.getEventProcessorConfig().isPresent()) {
                 log.info("Awaiting start of controller event processors");
-                controllerEventProcessors.awaitRunning();
+                // if store client has failed because of session expiration, there are two possibilities where 
+                // controllerEventProcessors.awaitRunning may be stuck forever -
+                // 1. stream creation is retried indefinitely and cannot complete because of zk session expiration
+                // 2. event writer after stream creation throws exception. 
+                // In both of above cases controllerEventProcessors.startAsync may not get called. 
+                CompletableFuture.anyOf(storeClientFailureFuture, 
```

Comment:
If the zk session expires on startup, then `storeClientFailureFuture` completes exceptionally and unblocks this call. However, if case (2) happens (event writer after stream creation throws exception), then `storeClientFailureFuture` does not complete and from the comment `controllerEventProcessors.startAsync` may not get called. Are we stuck in this case?

Reply:
No.
we are doing the following:
```
eventProcFuture = eventProcessors.bootStrap().thenAccept(v -> eventProcessors.startAsync());
...

CompletableFuture.anyOf(soreClientFailureFuture, eventProcFuture);
```

bootstrap:
1. IndefiniteRetry(createScope)
2. IndefiniteRetry(createStream)
3. CreateWriters

If `3` throws any exception then the future returned by bootStrap completes exceptionally. 
So `eventProcFuture` fails exceptionally with the same exception. 
And CompletableFuture.anyOf fails causing the startup to fail.